wandb/sdk/wandb_init.py

```diff
@@ -1281,15 +1281,32 @@ def try_create_root_dir(settings: Settings) -> None:
             if the parent directory is not writable.
     """
     try:
-        if not os.path.exists(settings.root_dir):
-            os.makedirs(settings.root_dir, exist_ok=True)
+        root_dir = settings.root_dir
+
+        if os.path.exists(root_dir):
+            if not os.access(root_dir, os.R_OK):
+                root_dir = tempfile.gettempdir()
+                wandb.termwarn(
+                    f"Path {settings.root_dir} wasn't readable, using system temp directory {root_dir}.",
+                    repeat=False,
+                )
+            elif not os.access(root_dir, os.W_OK):
+                root_dir = tempfile.gettempdir()
+                wandb.termwarn(
+                    f"Path {settings.root_dir} wasn't writable, using system temp directory {root_dir}.",
+                    repeat=False,
+                )
```

Comment:
Can combine all of these to shorten:

```python
if os.path.exists(root_dir) and not os.access(root_dir, os.R_OK | os.W_OK):
    root_dir = tempfile.gettempdir()
    wandb.termwarn(
        (
            f"Path {settings.root_dir} wasn't read-writable,"
            f" using system temp directory {root_dir}"
        ),
        repeat=False,
    )
```

It is also more readable to only set `root_dir` once:

```python
if ...:
    root_dir = tempfile.gettempdir()
else:
    root_dir = settings.root_dir
```

And once you do that, it's clear that you don't even need a root_dir variable. It looks like the `else` case just sets `settings.root_dir = settings.root_dir`. So you can drop the else case and the variable, and just set `settings.root_dir = tempfile.gettempdir()` in the if statement.